ssm_jax/hmm/models/gaussian_hmm.py

```diff
@@ -174,10 +169,10 @@ def _m_step_emissions(self, batch_emissions, batch_posteriors, **kwargs):
                                          mean_concentration=self._emission_prior_conc.value,
                                          df=self._emission_prior_df.value,
                                          scale=self._emission_prior_scale.value)
-        
+
         # Find the posterior parameters of the NIW distribution
         def _single_m_step(sum_w, sum_x, sum_xxT):
-            niw_posterior = niw_posterior_update(niw_prior, (sum_xxT, sum_x, sum_w))
+            niw_posterior = niw_posterior_update(niw_prior, (sum_x, sum_xxT, sum_w))
```

Comment:
This is the correct order for sufficient statistics.
I suppose you tested this part with the posterior update function without the integer conversion (as we discussed earlier)? My next PR reverts the posterior update function to the previous version without integer conversion, so this should be fine :)

Reply:
According to https://github.com/probml/ssm-jax/blob/main/ssm_jax/distributions.py#L280, the suff stats should be in the order ` Sx, SxxT, N = sufficient_stats`, so the new version looks correct to me.

Reply:
@xinglong-li As we discussed before, the integer conversion is not convenient to the type of the degrees of freedom of [tfp.distributions.WishartTriL](https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/WishartTriL) and it has made some tests to fail. So, I already reverted it here because we've settled on this way. 

@murphyk I believe the wrong order of sufficient statistics was the cause of the bug you mentioned at #166 If it is okay for you, could we merge it?

Reply:
I’m a bit behind on this, but a few students from the summer course have been hitting the integer conversion bug. It’d be great to prioritize a fix for that.